synapse/storage/deviceinbox.py

```diff
@@ -182,3 +238,64 @@ def get_all_new_device_messages_txn(txn):
 
     def get_to_device_stream_token(self):
         return self._device_inbox_id_gen.get_current_token()
+
+    def get_new_device_msgs_for_remote(
+        self, destination, last_stream_id, current_stream_id, limit=100
+    ):
+        """
+        Args:
+            destination(str): The name of the remote server.
+            last_stream_id(int): The last position of the device message stream
+                that the server sent up to.
+            current_stream_id(int): The current position of the device
+                message stream.
+        Returns:
+            Deferred ([dict], int): List of messages for the device and where
+                in the stream the messages got to.
+        """
```

Comment:
This probably wants to be behind a stream change cache to avoid hitting the DB for every TXN we want to send.